### ISSUE TITLE
shm provider does not work inside a container. 

### ISSUE THREAD

**Issue by wzamazon** [ofiwg/libfabric#5495]
I encountered this issue while try to use shm provider inside a container.

I used the following docker file to build my image:

```
FROM amazonlinux:1
ENV USER wzamazon 

RUN yum update -y 
RUN yum install -y bzip2 which gcc-c++ libtool autoconf automake util-linux make tar.x86_64 iproute2 gcc-gfortran openssh-server openssh-client python27 python27-pip build-essential wget curl git wget pkg-config zip unzip libnl3-devel
RUN pip-2.7 install supervisor 

RUN useradd -ms /bin/bash $USER
ENV HOME /home/$USER

RUN wget https://github.com/ofiwg/libfabric/archive/v1.9.0.tar.gz \
    && tar -zxf v1.9.0.tar.gz \
    && cd libfabric-1.9.0 \
    && ./autogen.sh \
    && ./configure --prefix=/opt/libfabric --disable-verbs --disable-usnic --disable-rstream --disable-rxd \
    && make \
    && make install

RUN wget https://github.com/ofiwg/libfabric/releases/download/v1.9.0/fabtests-1.9.0.tar.bz2 \
    && tar -jxf fabtests-1.9.0.tar.bz2 \
    && cd fabtests-1.9.0 \
    && ./configure --prefix=/opt/fabtests --with-libfabric=/opt/libfabric \
    && make \
    && make install
```

and use command:
```
docker build . -t libfabric
```
to build the image. then started a container using the following command:
```
docker run -it --user wzamazon --entrypoint /bin/bash libfabric
```
Inside container, I run the following command to test shm:

```
fi_rdm_pingpong -p shm -s 127.0.0.1 &
fi_rdm_pingpong -p shm -s 127.0.0.1 127.0.0.1 
```
and the output I got was:

```
 bytes   iters   total       time     MB/sec    usec/xfer   Mxfers/sec
bytes   iters   total       time     MB/sec    usec/xfer   Mxfers/sec
64      10k     1.2m        0.04s     34.86       1.84       0.54
64      10k     1.2m        0.04s     34.86       1.84       0.54
256     10k     4.8m        0.05s    107.39       2.38       0.42
256     10k     4.8m        0.05s    107.39       2.38       0.42
1k      10k     19m         0.05s    403.13       2.54       0.39
1k      10k     19m         0.05s    403.13       2.54       0.39
4k      10k     78m         0.08s   1063.62       3.85       0.26
4k      10k     78m         0.08s   1063.63       3.85       0.26
[error] fabtests:common/shared.c:2395: cq_readerr (Provider errno: 1) : Unknown error -1
[error] fabtests:common/shared.c:2395: cq_readerr (Provider errno: 1) : Unknown error -1
```

**Comment by wzamazon** [comment]
When I run the same command with FI_LOG_LEVEL=warn. I got an message:

libfabric:23:shm:ep_ctrl:smr_progress_iov():182<warn> CMA write error

Apparently, docker containers by default turns off ptrace, which CMA rely on.

In shm provider, there is a function smr_get_ptrace_scape() which read  file /proc/sys/kernel/yama/ptrace_scope, and determine ptrace support level, that file is not available in docker containers.

I think we need better mechanism to determine CMA support in shm provider.

**Comment by zhngaj** [comment]
Instead of checking the value in /proc/sys/kernel/yama/ptrace_scope, I think we can test with a CMA systemcall like the following.
```
nread = process_vm_readv(pid, local, 2, remote, 1, 0);
```
And check the errno (strerror(errno) will be `Operation not permitted`)

I feel this is a more straightforward and general approach.

**Comment by zhngaj** [comment]
PR #5510

**Comment by aingerson** [comment]
Fixed by #5510 and #5614
